annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -584,13 +593,21 @@ private SolrQuery solrUniquenessQuerySemanticTagging(Annotation anno, boolean no
 	
 	private SolrQuery solrUniquenessQuerySimpleTagging(Annotation anno, boolean noSelfDupplicate) {
       SolrQuery query = new SolrQuery();
-      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"");
-      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      query.setQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      if(anno.getTarget()!=null) {
+        List<String> targetValues = extractUriValues(anno.getTarget());
+        if(targetValues!=null) {
+          for(String target: targetValues) {
+            query.addFilterQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + target + "\"");
+          }
+        }
+      }
+      String bodyText = extractTextValues(anno.getBody());
       if(anno.getBody().getLanguage()!=null) {
-        query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":\"" + anno.getBody().getValue() + "\""); 
+        query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":\"" + bodyText + "\""); 
       }
       else {
-        query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + ":\"" + anno.getBody().getValue() + "\"");
+        query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE + ":\"" + bodyText + "\"");
```

Comment:
it must be the SolrAnnotationConstants.BODY_VALUE_PREFIX (the solr field is body_value.)

Reply:
Improved.